tests/shared/crclient/client.go

```diff
@@ -124,12 +142,22 @@ func (f Client) listPodList(list *corev1.PodList) error {
 	return nil
 }
 
-func (f Client) listNodeList(list *corev1.NodeList) error {
+func (f Client) listNodeList(list *corev1.NodeList, label string) error {
+	_, log := logger.GetNewContextWithLogger("0")
 	for k, v := range f.Objects {
 		if k.Kind == "Node" {
-			node, ok := v.(*corev1.Node)
-			if ok {
-				list.Items = append(list.Items, *node)
+			node := *v.(*corev1.Node)
+			if label != "" {
+				for key := range node.ObjectMeta.Labels {
+					if label == key {
+						log.Infof("\tadding node name:%v to list matching label key \n", node.Name)
```

Comment:
`\n` is redundant. Also, matching which key? Do we need this debug log here at all?

Reply:
removed.